raven/src/intTest/java/org/opencadc/raven/NegotiationTest.java

```diff
@@ -180,12 +176,81 @@ public Object run() throws Exception {
                     }
                 }
             });
-            
-        } catch (Throwable t) {
-            log.error("unexpected throwable", t);
-            Assert.fail("unexpected throwable: " + t);
-        } finally {
-            System.clearProperty(PropertiesReader.CONFIG_DIR_SYSTEM_PROPERTY);
+        } catch (Exception e) {
+            log.error("unexpected exception", e);
+            Assert.fail("unexpected exception: " + e);
+        }
+    }
+
+    @Test
+    public void testPUT() {
+        try {
+            Subject.doAs(userSubject, new PrivilegedExceptionAction<Object>() {
+                public Object run() throws Exception {
+
+                    URI resourceID1 = URI.create("ivo://negotiation-test-site1");
+                    URI resourceID2 = URI.create("ivo://negotiation-test-site2");
+
+                    ArtifactDAO artifactDAO = new ArtifactDAO();
+                    artifactDAO.setConfig(config);
+                    StorageSiteDAO siteDAO = new StorageSiteDAO();
+                    siteDAO.setConfig(config);
+                    StorageSite site1 = new StorageSite(resourceID1, "site1", true, true);
+                    StorageSite site2 = new StorageSite(resourceID2, "site2", true, false);
+
+                    URI artifactURI = URI.create("cadc:TEST/" + UUID.randomUUID() + ".fits");
+                    URI checksum = URI.create("md5:d41d8cd98f00b204e9800998ecf8427e");
+                    Artifact artifact = new Artifact(artifactURI, checksum, new Date(), 1L);
+
+                    try {
+                        siteDAO.put(site1);
+                        siteDAO.put(site2);
+
+                        SiteLocation location1 = new SiteLocation(site1.getID());
+                        SiteLocation location2 = new SiteLocation(site2.getID());
+
+                        Protocol protocol = new Protocol(VOS.PROTOCOL_HTTPS_PUT);
+                        Transfer transfer = new Transfer(
+                            artifactURI, Direction.pushToVoSpace, Arrays.asList(protocol));
+                        transfer.version = VOS.VOSPACE_21;
```

Comment:
To be complete, the transfer negotiation test should ask for multiple protocols:
https anon (no securityMethod -- the one that's there)
https w/ cert
https w/ cookie
https w/ token
http anon

The correct response would include:
1x protocol(https URL with pre-auth, securityMethod null) aka anon per site
3x protocol(https URl to minoc, securityMethod = {credential type}) per site
no http protocol (because minoc doesn't support http)

In principle, the protocols w/ securityMethod may or may not include the pre-auth stuff. TBD  so don't change whatever is there. All of this applies to GET and PUT.